### ISSUE TITLE
Give standalone users a "Shut Up" keybind

### ISSUE THREAD

**Issue by richardbuckle** [EDCD/EDDI#197]
While VA users can say "Shut Up" to stop EDDI talking, standalone users have no such option. 

Allow standalone users to configure a global keybind to invoke EDDI's `ShutUp()` function.

Ideally users should be able to define two keybinds, and DirectX inputs such as HOTAS, game pads etc should be available as well as the keyboard.

NB don't make the mistake that Discord did: store the device as well as the button ID. Just because I bound button 9 on my stick doesn't mean that I want button 9 on my throttle bound too ;)

**Comment by Tkael** [comment]
VA users actually have 3 commands that may be relevant here. One to shut EDDI up, one to disable the speech responder, and a third to enable the speech responder. If we take this route, we would probably want to make all three available for keybinding.

**Comment by richardbuckle** [comment]
I would settle for just the one that means "Shut up what you're saying right now".

**Comment by Tkael** [comment]
I wonder... would fleshing out the script priority system in EDDI accomplish the same purpose in your opinion?

**Comment by richardbuckle** [comment]
I don’t think it would. The `ShupUp()` function interrupts EDDI mid-sentence which is not quite the same thing.

**Comment by Tkael** [comment]
The `SharpDX` library (or similar) looks like it will be needed for DirectX inputs. http://sharpdx.org/

**Comment by richardbuckle** [comment]
Good find. Let’s also ask the other EDCD folks for recommendations.

**Comment by richardbuckle** [comment]
I'd be happy with just a customisable keyboard input for a first iteration. We can look at DirectX later.

**Comment by shodanx2** [comment]
Hello, 
This issue is probably making most EDDI users just disable all text to voice on their first session.
How does VA tell EDDI to "shutup"  ?
Surely that can be emulated with random hotkey software like AutoIT ?

Or maybe from a command line like  EDDI.exe -shutup  and just invoke that from AutoIT.

**Comment by Tkael** [comment]
At the moment, it is not possible to do this from either a hotkey or from the command line. The VoiceAttack plugin command directly invokes a method in the SpeechService class.
https://github.com/EDCD/EDDI/blob/79121ebfabaee879c35961cc164a9124cdb24aef/VoiceAttackResponder/VoiceAttackPlugin.cs#L872-L886

**Comment by shodanx2** [comment]
Could AutoIT be used to trigger this ? 
I just started playing ED, first thing I did after re-installing EDDI is turn off text to speech
Which is a shame because I would like it.
But I can't wait for it to shut up if I need to talk to my friend or if they're talking to me.

**Comment by Tkael** [comment]
Not that I'm aware of (I don't use AutoIT).

**Comment by Tkael** [comment]
Merged in PR #2738.
